apps/web/src/services/dashboards/components/dashboard-detail/DashboardGlobalVariableFilterReference.vue

```diff
@@ -0,0 +1,220 @@
+<script setup lang="ts">
+
+
+import { computed, reactive, watch } from 'vue';
+
+import { cloneDeep, flattenDeep } from 'lodash';
+
+import { PSelectDropdown } from '@cloudforet/mirinae';
+import type { MenuItem } from '@cloudforet/mirinae/types/inputs/context-menu/type';
+import type { AutocompleteHandler } from '@cloudforet/mirinae/types/inputs/dropdown/select-dropdown/type';
+
+import type {
+    DashboardGlobalVariable,
+    ReferenceVariable,
+} from '@/schema/dashboard/_types/dashboard-global-variable-type';
+import type { DashboardVars } from '@/schema/dashboard/_types/dashboard-type';
+import type { WorkspaceModel } from '@/schema/identity/workspace/model';
+
+import { useUserWorkspaceStore } from '@/store/app-context/workspace/user-workspace-store';
+
+import { VariableModelFactory } from '@/lib/variable-models';
+import { MANAGED_VARIABLE_MODELS } from '@/lib/variable-models/managed-model-config/base-managed-model-config';
+import {
+    getVariableModelMenuHandler,
+} from '@/lib/variable-models/variable-model-menu-handler';
+
+import WorkspaceLogoIcon from '@/common/modules/navigations/top-bar/modules/top-bar-header/WorkspaceLogoIcon.vue';
+
+import { getWorkspaceInfo } from '@/services/advanced/composables/refined-table-data';
+import { useDashboardDetailInfoStore } from '@/services/dashboards/stores/dashboard-detail-info-store';
+
+interface Props {
+    variable: DashboardGlobalVariable;
+}
+
+
+const props = defineProps<Props>();
+const userWorkspaceStore = useUserWorkspaceStore();
+const workspaceStoreGetters = userWorkspaceStore.getters;
+const dashboardDetailStore = useDashboardDetailInfoStore();
+const dashboardDetailState = dashboardDetailStore.state;
+
+const storeState = reactive({
+    workspaceList: computed<WorkspaceModel[]>(() => workspaceStoreGetters.workspaceList),
+    vars: computed<DashboardVars>(() => dashboardDetailState.vars),
+});
+
+const state = reactive({
+    variable: computed(() => {
+        const referenceVariable = props.variable as ReferenceVariable;
+        return referenceVariable;
+    }),
+    multiSelectable: computed(() => state.variable.options.selectionType === 'multi'),
+    menuHandler: computed<AutocompleteHandler>(() => {
+        const listQueryOptions: Record<string, any> = {};
+        const targetModelConfig = Object.values(MANAGED_VARIABLE_MODELS).find((d) => (d.meta?.resourceType === state.variable.reference.resourceType));
+
+        const option = {
+            type: 'MANAGED',
+            key: targetModelConfig?.meta.key,
+            dataKey: state.variable.reference.dataKey,
+        };
```

Comment:
nitpick:
Since it’s wrapped in a variable called `option`, it gives the impression that it might be used elsewhere, but that’s not the case. If it’s just for convenience, it might be better to define each one as an individual variable.

이렇게 option이라는 변수로 묶이니까, 어딘가에 따로 쓰일 것 같지만 그렇지 않네요.
편의성을 위한 것이라면 그냥 각각을 변수로 만드는 게 나을 것 같아요.